tests/functional/conftest.py

```diff
@@ -401,6 +401,21 @@ def _merge_request(*, source_branch: str):
                 "commit_message": "New commit in new branch",
             }
         )
+        if create_pipeline:
+            project.files.create(
+                {
+                    "file_path": ".gitlab-ci.yml",
+                    "branch": source_branch,
+                    "content": """
+test:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
+  script:
+    - sleep 24h  # We don't expect this to finish
+""",
+                    "commit_message": "Add a simple pipeline",
+                }
+            )
```

Comment:
One thought I had here, to keep the MR fixture generic/short we could also create a specific `merge_request_pipeline` or `merge_request_with_pipeline` that uses `merge_request`, the same way we do with `project_file` when we need a project that also has a file (https://github.com/python-gitlab/python-gitlab/blob/main/tests/functional/conftest.py#L432-L444). Then in tests we can just choose which fixture to use.

Although maybe the pattern is a bit different here :thinking:

Reply:
Sure, I can do that.

Reply:
Okay. I investigated doing that and it got complicated. More complicated, IMHO, then having the argument to the function.

As the `merge_request_with_pipeline` would need to replicate a large portion of `merge_request`

Reply:
A python question, is it possible to use one fixture in a second fixture? Could that work?

Reply:
Ok thanks, let's not complicate :) @lmilbaum yes, generally that was my idea. But this fixture here is more like a [fixture factory](https://docs.pytest.org/en/6.2.x/fixture.html#factories-as-fixtures) so a slightly different approach as you call it in the test case, so got a bit complicated. We can split them into `merge_request` and `make_merge_request` at some point if we want that kind of reuse eventually.